content/posts/2015/08/2015-08-14-secure-central-hosting-for-the-digital-analytics-program.md

```diff
@@ -49,28 +51,30 @@ The [recent federal government policy on HTTPS](https://https.cio.gov/) requires
 
 Because dap.digitalgov.gov is a potential high-value target, the Digital Analytics Program did not want to support plain HTTP connections at all—even for a redirect. Though HTTP redirects are helpful, they are still an opportunity for attack. [HSTS](https://https.cio.gov/hsts/) is designed to help with this, but an even more secure solution is to simply disable HTTP altogether.
 
 This generally isn&#8217;t a viable solution for websites, because users type bare domains like &#8220;whitehouse.gov&#8221; into browser location bars, and browsers generally have to assume plain HTTP as a first try in these situations. But because dap.digitalgov.gov is a brand new subdomain used only as a third party service, DAP can set a higher standard by **breaking the protocol-relative URL** when used on a plain HTTP site.
 
-The simplest solution is to refuse http:// connections entirely by closing port 80 and not allowing browsers to connect at all, but this was not a viable option for DAP&#8217;s hosting provider. However, returning an error code instead of a redirect for plain HTTP connections would not be in compliance with federal HTTPS policy.
+The simplest solution is to refuse `http://` connections entirely by closing port 80 and not allowing browsers to connect at all, but this was not a viable option for DAP&#8217;s hosting provider. However, returning an error code instead of a redirect for plain HTTP connections would not be in compliance with federal HTTPS policy.
 
 We solved the issue by **combining** a redirect with an error code. Any HTTP requests to a file on http://dap.digitalgov.gov will redirect the user to https://dap.digitalgov.gov/403, which then returns a 403 error code.
 
-<div>
-  <code>$ curl --head http://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js&lt;br />
-HTTP/1.1 301 Moved Permanently&lt;br />
-Location: https://dap.digitalgov.gov/403</code>
-</div>
+```
+$ curl --head http://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js
+HTTP/1.1 301 Moved Permanently
+Location: https://dap.digitalgov.gov/403
+```
 
 This ensures that data can _only_ be collected over HTTPS, and _breaks_ any HTTP or protocol-relative URLs participating agencies might accidentally use when integrating their websites into the Digital Analytics Program.
 
 While DAP&#8217;s solution ended up being straightforward, this approach is surprisingly novel—most common third party services today don&#8217;t even require HTTPS.
 
-But on today&#8217;s Internet, they should, and the Digital Analytics Program is leading by example._Eric Mill is an 18f team member._
\ No newline at end of file
+But on today&#8217;s Internet, they should, and the Digital Analytics Program is leading by example.
+
+_Eric Mill is an 18f team member._
```

Comment:
Teensy nit: probably was this way at the time, but capitalizing the `F` in `18F` would be more to-style. :)